flutter.el

```diff
@@ -118,17 +120,52 @@ ARGS is a space-delimited string of CLI flags passed to
           (flutter-mode)))
       ,@body)))
 
+(defmacro flutter--with-test-proc (args &rest body)
+  "Execute BODY while ensuring an inferior `flutter test` process is running.
+
+BUFFER-NAME is the chosen buffer where will run the proc.
+ARGS is a space-delimited string of CLI flags passed to
+`flutter`, and can be nil."
+  `(flutter--from-project-root
+    (let* ((buffer (get-buffer-create flutter-test-buffer-name))
+           (alive (flutter--testing-p))
+           (arglist (when ,args (split-string ,args))))
+      (unless alive
+        (apply #'make-comint-in-buffer "Flutter Test" buffer (flutter-build-command) nil "test" arglist))
+      (with-current-buffer buffer
+        (unless (derived-mode-p 'flutter-mode)
+          (flutter-mode)))
+      ,@body)))
```

Comment:
This is a copy-and-paste of `flutter--with-run-proc`. Copy-and-paste is not bad in and of itself, but here I don't think it's appropriate:

`flutter--with-run-proc` is the way it is because `flutter run` is a long-running process that you want to send commands to. That's why there is all this machinery to make it easy to run things in an already extant buffer.

For test running, I expect it to be a one-shot execution that you do not care to communicate with. For this you don't need all of this machinery. I suggest you take a look at [`ruby-test-run` and `ruby-test-run-at-point` in ruby-test-mode](https://github.com/ruby-test-mode/ruby-test-mode/blob/e805a81c3ea787f3c82dbb252c21d747be20bc7a/ruby-test-mode.el#L359-L384).

Reply:
Nice! I'll fix it!